pkg/pillar/cmd/client/client.go

```diff
@@ -201,13 +201,7 @@ func Run(ps *pubsub.PubSub) { //nolint:gocyclo
 	clientCtx.subDeviceNetworkStatus = subDeviceNetworkStatus
 	subDeviceNetworkStatus.Activate()
 
-	zedcloudCtx := zedcloud.ZedCloudContext{
-		DeviceNetworkStatus: clientCtx.deviceNetworkStatus,
-		FailureFunc:         zedcloud.ZedCloudFailure,
-		SuccessFunc:         zedcloud.ZedCloudSuccess,
-		NetworkSendTimeout:  clientCtx.globalConfig.NetworkSendTimeout,
-		V2API:               zedcloud.UseV2API(),
-	}
+	zedcloudCtx := zedcloud.NewContext(clientCtx.deviceNetworkStatus, clientCtx.globalConfig.NetworkSendTimeout, true)
```

Comment:
Instead of a fixed set of arguments, it is clearer and more flexible to define a zedcloud.ContextOptions struct and then intialize it with named fields in the call e.g.
 zedcloud.NewContext(
     zedcloud.ContextOptions{
            DeviceNetworkStatus: clientCtx.deviceNetworkStatus,
            ...

Reply:
ok, i'll change to that option thing.